### ISSUE TITLE
FusionPBX Domain Routing Sync Causes Kam to Run Out of Memory

### ISSUE THREAD

**Issue by mackhendricks** [dOpensource/dsiprouter#193]
We need to rewrite the logic for doing the sync so that only changed Domains are modified versus removing the entire domain table and reloading it.

**Comment by devopsec** [comment]
We may be able to get away with using `insert ignore` and `delete from domain where did=`
See: https://github.com/dOpensource/dsiprouter/blob/v0.523/gui/modules/fusionpbx/fusionpbx_sync_functions.py#L119

the code in that file is really hard to sift through we may want to push this back.

**Comment by mackhendricks** [comment]
The real issue is that the domain module in Kamailio 5.1.8 has a memory leak.  We use this module to keep track of Domains and attributes about that domain.  We have reported the fix to the Kamailio dev team and they have already applied a fix.  However, we aren't sure when this will make it into main branches.  

Therefore, we created a work around that should minimize (low down) the memory leak problem.  The workaround consist of creating a hash of the domain names that we sync.  Hence, we only sync with FusionPBX when a domain is added or deleted from FusionPBX.  This means that we only reload the domain module when an actually change occurs versus us reloading it even if a change hasn't occurred.
